src/dom/dom.ts

```diff
@@ -980,10 +1025,77 @@ export class dom {
         allTextNodes.forEach((node)=> {
            node.nodeValue = node.nodeValue.replace(/\s+([^\s]*)\s*$/, nbsp + '$1')
         })
+    }
 
+    /**
+     * Add an event listener to multiple events.
+     * Useful for when mouse and touch events should call the same listener.
+     */
+    static addEventListeners(
+        element: Element|Window,
+        events: string[],
+        listener: EventListenerOrEventListenerObject
+    ) {
+        events.forEach(
+            (event) => element.addEventListener(event, listener));
+    }
+
+    /**
+     * Remove an event listener for multiple events.
+     * Useful for when mouse and touch events should call the same listener.
+     */
+    static removeEventListeners(
+        element: Element|Window,
+        events: string[],
+        listener: EventListenerOrEventListenerObject
+    ) {
+        events.forEach(
+            (event) => element.removeEventListener(event, listener));
     }
 
     static getScrollElement(): Element {
         return document.scrollingElement || document.documentElement;
     }
+
+    /**
+     * Determine the distance between the centers of two given elements.
+     * If no second element is given, the root element is used.
+     *
+     * Uses the matrix service to account for translations that will be applied
+     * during the raf write step. Note that MatrixService can only account for
+     * translations that will be applied via MatrixService.
+     *
+     * @param a
+     * @param b
+     */
+    static getVisibleDistanceBetweenCenters(
+        a: HTMLElement, b: HTMLElement = null
+    ): Vector {
+        // Gather up the information on the first element's center position.
+        const aRect = a.getBoundingClientRect();
```

Comment:
I'm a bit confused why we need MatrixService here given that getBoundingClientRect accounts for translation positions.  Do we need this?

Reply:
I'll move this to within the carousel. Sorry, at one point MatrixService lived outside of Carousel, when it was consolidated this was missed.

The reason MatrixService is used is so that when this function is queried it takes into account translations that are scheduled to be applied during the next write step. Without this some effects become out of sync and jitter.